migrations/2023-08-13-175220_to_semver_no_prerelease_parallel_safe/up.sql

```diff
@@ -0,0 +1,17 @@
+DROP FUNCTION to_semver_no_prerelease(text);
+DROP TYPE semver_triple;
+
+CREATE TYPE semver_triple AS (
+  major numeric,
+  minor numeric,
+  teeny numeric
+);
+
+CREATE FUNCTION to_semver_no_prerelease(text) RETURNS semver_triple IMMUTABLE PARALLEL SAFE AS $$
+  SELECT (
+    split_part($1, '.', 1)::numeric,
+    split_part($1, '.', 2)::numeric,
+    split_part(split_part($1, '+', 1), '.', 3)::numeric
+  )::semver_triple
+  WHERE strpos($1, '-') = 0
+  $$ LANGUAGE SQL;
```

Comment:
This isn't strictly related to what you're doing, but while we're in here, we should also add:

```suggestion
  $$ LANGUAGE SQL SET search_path = public;
```

Right now this works by accident: the `schema.sql` generated when we dump the database each night is actually subtly wrong, but PostgreSQL doesn't check the actual return type of the function definition unless it's used somewhere, which it's not when importing the data dump. However, if and when we add a generated column (as in #6910), this will start getting used, and an error occurs when importing the schema because the return type isn't `public.semver_triple`.

Reply:
CI doesn't seem to like it

Reply:
It doesn't! Sorry about that.

I hope you don't mind — I've pushed a commit that should fix it. (Easier than trying to do it through GitHub code review suggestions.)